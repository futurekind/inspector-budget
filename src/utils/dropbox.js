const KEY = 'IB_AT';
const APP_KEY = '4inxggnvthsk657';
const AUTHORIZE_URL = 'https://www.dropbox.com/1/oauth2/authorize'

export const shouldGetAccessToken = () => {
    if(localStorage.getItem(KEY)) {
        return false
    } else {
        return true
    }
}

export const authorize = () => {
    window.location.href = `${AUTHORIZE_URL}?response_type=token&client_id=${APP_KEY}&redirect_uri=http://localhost:3000/`
}

export const getAccessTokenInHashFragment = hashStr => {
    const fragments = hashStr.split('&').reduce((comp, fragment) => {
        const piece = fragment.split('=');
        return Object.assign({}, comp, {
            [piece[0]]:piece[1]
        });
    }, {});
    
    return fragments.access_token || false;
}

export const setAccessToken = token => {
    localStorage.setItem(KEY, token);
}

export const getAccessToken = () => localStorage.getItem(KEY);