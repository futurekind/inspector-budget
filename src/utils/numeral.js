import numeral from 'numeral';
import de from 'numeral/locales/de';

numeral.defaultFormat('0,0.00')
numeral.locale('de');

export { numeral };