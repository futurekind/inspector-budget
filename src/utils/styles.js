export const colors = {
    black: '#000',
    dark: '#333',
    light: '#888'
}

export const sizes = {
    navigationWidth: 90
}