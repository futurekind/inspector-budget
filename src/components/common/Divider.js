import React, {PropTypes} from 'react';
import styled from 'styled-components';
import { colors } from '../../utils/styles'

const Hr = styled.hr`
    width: 100%;
    height: 1px;
    border: none;
    background-color: ${props => props.color}
`
const getColor = type => {
    switch (type) {
        case 'light':
            return colors.light;
        case 'highlight':
            return colors.highlight
        default:
            return colors.dark
    }
}

const Divider = ({
    type
}) => {
    return (
        <Hr color={ getColor(type) } />
    )
}

Divider.defaultProps = {
    type: 'default'
}

Divider.propTypes = {
    type: PropTypes.oneOf([
        'default', 'light', 'highlight'
    ]),
}

export default Divider