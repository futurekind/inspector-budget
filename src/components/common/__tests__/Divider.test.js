import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Divider from '../Divider';
import { colors } from '../../../utils/styles'

describe('Divider', () => {
    const wrapper = shallow(<Divider />);

    it('renders without errors', () => {
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders the default color', () => {
        expect(wrapper.props().color).toBe(colors.dark)
    })

    it('renders type="light"', () => {
        wrapper.setProps({
            type: 'light'
        })

        expect(wrapper.props().color).toBe(colors.light)
    })

    it('renders type="highlight"', () => {
        wrapper.setProps({
            type: 'highlight'
        })

        expect(wrapper.props().color).toBe(colors.highlight)
    })

})