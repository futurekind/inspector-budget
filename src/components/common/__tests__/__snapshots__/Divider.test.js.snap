exports[`Divider renders without errors 1`] = `
<styled.hr
  color="#333" />
`;
